project/resources/js/Components/DataList.vue

```diff
@@ -1,22 +1,40 @@
+<template>
+    <div>
+        <slot name="options">
+        </slot>
+
+        <table class="col-12">
+            <thead>
+                <slot name="header" :orderBy="orderBy"></slot>
+            </thead>
+            <tbody>
+                <slot name="body" :fetchData="fetchData" :items="items"></slot>
+            </tbody>
+        </table>
+
+        <pagination
+          class="col-10"
+          :total-pages="totalPages"
+          :current-page="currentPage"
+          @fetchPrevPage="fetchPrevPage"
+          @fetchNextPage="fetchNextPage"
+          @changePage="page => changePage(page)">
+        </pagination>
+    </div>
+</template>
+
 <script>
   import SortIcon from '../support/SortIcon.js';
+  import Pagination from "./Pagination";
 
   export default {
-    template: '#data-list',
-
-    props: {
+      components: {Pagination},
+      props: {
       dataSource: {
         type: String,
       },
     },
 
-    watch: {
```

Comment:
Why did you remove this?

Reply:
The text filter is in the text component. Debounce is more appropriate there.